src/plugins/internalnotification/internalnotification.c

```diff
@@ -0,0 +1,263 @@
+		fprintf (stderr, "elektraInternalnotificationUpdateRegisteredKeys looking up registeredKey=%s\n", registeredKey->name);
+#endif
+
+		Key * key = ksLookupByName (keySet, registeredKey->name, 0);
+		if (key != 0)
+		{
+#if PLUGIN_INTERNALNOTIFICATION_VERBOSE
+			fprintf (stderr,
+				 "elektraInternalnotificationUpdateRegisteredKeys found registeredKey=%s; updating variable=%p with string "
+				 "value %s\n",
+				 registeredKey->name, (void *)registeredKey->variable, keyString (key));
+#endif
+			// Convert string value to long
+			char * end;
+			errno = 0;
+			long int value = strtol (keyString (key), &end, 10);
+			// Update variable if conversion was successful and did not exceed integer range
+			if (*end == 0 && errno == 0 && value <= INT_MAX && value >= INT_MIN)
+			{
+				*(registeredKey->variable) = value;
+			}
+#if PLUGIN_INTERNALNOTIFICATION_VERBOSE
+			else
+			{
+				fprintf (stderr,
+					 "elektraInternalnotificationUpdateRegisteredKeys conversion failed! keyString=%s, *end=%c, "
+					 "errno=%d, value=%ld\n",
+					 keyString (key), *end, errno, value);
+			}
+#endif
+		}
+
+		registeredKey = registeredKey->next;
+	}
+}
+
+int elektraInternalnotificationGet (Plugin * handle, KeySet * returned, Key * parentKey)
+{
+	const char * parentKeyName = keyName (parentKey);
+
+	if (!elektraStrCmp (parentKeyName, "system/elektra/modules/internalnotification"))
+	{
+		KeySet * contract = ksNew (
+			30, keyNew ("system/elektra/modules/internalnotification", KEY_VALUE,
+				    "internalnotification plugin waits for your orders", KEY_END),
+			keyNew ("system/elektra/modules/internalnotification/exports", KEY_END),
+			keyNew ("system/elektra/modules/internalnotification/exports/get", KEY_FUNC, elektraInternalnotificationGet,
+				KEY_END),
+			keyNew ("system/elektra/modules/internalnotification/exports/set", KEY_FUNC, elektraInternalnotificationSet,
+				KEY_END),
+			keyNew ("system/elektra/modules/internalnotification/exports/open", KEY_FUNC, elektraInternalnotificationOpen,
+				KEY_END),
+			keyNew ("system/elektra/modules/internalnotification/exports/close", KEY_FUNC, elektraInternalnotificationClose,
+				KEY_END),
+
+			// Export registerInt function and required plugin handle
+			keyNew ("system/elektra/modules/internalnotification/exports/elektraInternalnotificationRegisterInt", KEY_FUNC,
+				elektraInternalnotificationRegisterInt, KEY_END),
+			keyNew ("system/elektra/modules/internalnotification/exports/handle", KEY_BINARY, KEY_SIZE, sizeof handle,
+				KEY_VALUE, &handle, KEY_END),
+
+#include ELEKTRA_README (internalnotification)
+
+			keyNew ("system/elektra/modules/internalnotification/infos/version", KEY_VALUE, PLUGINVERSION, KEY_END), KS_END);
+		ksAppend (returned, contract);
+		ksDel (contract);
+
+		return 1;
+	}
+#if PLUGIN_INTERNALNOTIFICATION_VERBOSE
+	fprintf (stderr, "elektraInternalnotificationGet keyName=%s\n", parentKeyName);
+#endif
+
+	ListPointer * listPointer = elektraPluginGetData (handle);
```

Comment:
I think you should check that the fetched data is not NULL and then cast properly (occurs several times).